rescheduling-service/src/services/appointmentService.ts

```diff
@@ -0,0 +1,29 @@
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+const fetchAvailableAppointments = async () => {
+  try {
+    const availableAppointmentsWithOffers = await prisma.appointment.findMany({
+      where: {
+        booked: false,
+      },
+      include: {
+        offer: true,
+      },
+      orderBy: {
+        date: 'asc',
+      },
+    });
+
+    const availableAppointments = availableAppointmentsWithOffers.filter(
+      (appointment) => !appointment.offer
+    );
+    return availableAppointments;
+  } catch (error) {
+    console.error('Error fetching available appointments', error);
+    throw new Error('Unable to fetch available appointments');
+  }
+};
```

Comment:
I think you should separate the logic connected with interactions with DB in a folder like 'db-service', create a file like AppointmentRepository, and collect all interactions related to this table, etc. You can check it in auth-svc.